### ISSUE TITLE
Use a fixed identity instead of generating one per person

### ISSUE THREAD

**Issue by haasn** [mmb/weechat-otr#70]
It would be cool if I could use a single identity with a fingerprint that can be uniquely tied to be, eg. via a business card or a NameCoin entry.

Of course, using a PGP key for this instead and signing my message with that is a valid fallback strategy - but given the fact that eg. NameCoin already supports including otr fingerprints in personal records it may as well be a support feature.

**Comment by Mikaela** [comment]
Is there any schedule on option to set default key? I am just wondering as I have multiple keys and I have no idea which of those [match my public keys page...](https://mikaela.info/keys/)

```
Anima@ch.key3  Anima@en.key3  Anima@pi.key3  Anima@qu.key3   Mikaela@bi.key3  Mikaela@fr.key3  Mikaela@ne.key3  Mikaela@pi.key3  Mikaela@qu.key3
Anima@en.fpr   Anima@pi.fpr   Anima@qu.fpr   Mikaela@bi.fpr  Mikaela@fr.fpr   Mikaela@ne.fpr   Mikaela@pi.fpr   Mikaela@qu.fpr
```

It would be a lot easier if I could copy the fr ones elsewhere, remove others and set default key to have future OTR discussions to use the same key.

**Comment by mmb** [comment]
I'll make this the next feature that I work on. In the meantime try /otr
fingerprint.

**Comment by Mikaela** [comment]
> In the meantime try /otr

Oh, I wasn't aware of this command. After removing my altnick keys (which will probably get very random next time someone starts OTR with me and I happen to have one…) it appears that I only have three networks where there is wrong key/missing one.

**Comment by haasn** [comment]
I've just realized: What are the possible security implications of this? Is there a good reason that weechat-otr was initially designed to generate a new identity per contact?

**Comment by Mikaela** [comment]
> Is there a good reason that weechat-otr was initially designed to generate a new identity per contact?

I think I saw reasoning somewhere saying that this was done to preserve anonymity so multiple identities cannot be associated to you based on your key.

But I have issue of wanting people to be able to verify that I am the same person across networks.

**Comment by Ghosthree3** [comment]
Is there any way to use the same key across different servers where you use the same _nick_ but not for all nicks on all servers? ie. If I'm using the name Ghosthree3 on multiple servers I clearly want to be recognised as that identity, so I would ideally like to use the same fingerprint for them all. But if I have another server where I'm using a different name with the clear intention of anonymising myself I would like to use a different fingerprint for that identity.

So basically I would like my keys to be on a nick basis instead of nick+server. Is that possible without manually copying keys around?

**Comment by mmb** [comment]
@Ghosthree3 There is currently no way to configure one key for all nicks on any server. The closest you could get is to set your default key and copy files for any nick@server that you didn't want to use your default key.
